### ISSUE TITLE
[ErrorHandler] default configuration doesn't seem to have any effect at all

### ISSUE THREAD

**Issue by kgkg** [symfony/symfony#35359]
**Symfony version(s) affected**: 4.4

**Description**  
When registering ErrorHandler::register() as in https://symfony.com/doc/current/components/error_handler.html doesn't seem to make any effect. Documentation is also missing examples of defining our own ErrorHandler.

**How to reproduce**  
use Symfony\Component\ErrorHandler\ErrorHandler;
ErrorHandler::register();
throw new \Exception('any expcetion');

**Expected behaviour**  
1) dev env: display exception/errors and log them in dev.log (current: displays only)
2) prod env: don't display exceptions/errors and log them in prod.log (current: displays only, same as in dev)

Using Debug::enable() make errors appear in dev.log but from what we can read here https://symfony.com/blog/new-in-symfony-4-4-errorhandler-component ErrorHandler should replace old Debug::enable(). There is also recommendation that we should not use Debug::register() in production.

**Comment by nicolas-grekas** [comment]
What are you trying to do? Calling a global static factory won't wire the handler with the other Symfony services - nothing is global in Symfony. But there is already a handler that is wired by default, so why call this?

**Comment by kgkg** [comment]
I can be pretty possible I misuse it - I have legacy system in Wordpress which I want to migrate into Symfony, but most of pages are still generated in Wordpress, so I just booted Symfony kernel and requested services from container (not using controllers request handling). 
I used ErrorHandler this way for 2 reasons:
1) it just worked that way with Debug::enable
2) ErrorHandler was announced to replace Debug, so I thoutght I use it the same way, and docs are seems to suggest using it that way.

I just wanted it to automatically set error/exception handlers depending on Symfony configuration. I guess this kind of use case is not supported by ErrorHandler alone.

Anyway I managed to configure monolog's ErrorHandler and disabling Symfony ErrorHandler to not render error pages on production so it's not issue to me right now.
